ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -437,17 +421,51 @@ private TransactionSelectionResult handleTransactionNotSelected(
 
     final var pendingTransaction = evaluationContext.getPendingTransaction();
 
-    transactionSelectionResults.updateNotSelected(
-        evaluationContext.getTransaction(), selectionResult);
-    pluginTransactionSelector.onTransactionNotSelected(evaluationContext, selectionResult);
+    final TransactionSelectionResult actualResult =
+        isTimeout.get() && !wasTimeout(selectionResult)
```

Comment:
The names of the conditions sound contradictory even though they might be what you want. The first one would mean there is a timeout but the next one would mean not a timeout. 

Maybe rename `wasTimeout` to `[in]correctTimeoutType`?

Reply:
I see what you mean, will find for a better wording, `!wasTimeout(selectionResult)` is there to identify if the timeout was already processed, in that case we do not need to reprocess it, so probably naming it `wasTimeoutAlreadyProcessed` reads better?

Reply:
Yes, definitely better

Reply:
rename done